apps/calendar/js/bridge.js

```diff
@@ -0,0 +1,118 @@
+define(function(require, exports) {
+'use strict';
+
+var EventEmitter2 = require('ext/eventemitter2');
+var co = require('ext/co');
+var core = require('core');
+var dayObserver = require('day_observer');
+
+/**
+ * Fetch all the data needed to display the busytime information on the event
+ * views based on the busytimeId
+ */
+exports.fetchRecord = function(busytimeId) {
+  return co(function *() {
+    var record = yield dayObserver.findAssociated(busytimeId);
+    var eventStore = core.storeFactory.get('Event');
+    var owners = yield eventStore.ownersOf(record.event);
+    var provider = core.providerFactory.get(owners.account.providerType);
+    var capabilities = yield provider.eventCapabilities(record.event);
+
+    record.calendar = owners.calendar;
+    record.account = owners.account;
+    record.capabilities = capabilities;
+
+    return record;
+  });
+};
+
+/**
+ * Fetch all the calendars from database and emits a new event every time the
+ * values changes.
+ *
+ * @returns {ClientStream}
+ */
+exports.listenCalendars = function() {
+  // TODO: replace with real threads.client.stream when we get db into worker
+  var stream = new FakeClientStream();
+  var calendarStore = core.storeFactory.get('Calendar');
+
+  var getAllAndWrite = co.wrap(function *() {
+    // calendarStore.all() returns an object! we convert into an array since
+    // that is easier to render/manipulate
+    var calendars = yield calendarStore.all();
+    var data = [];
+    for (var id in calendars) {
```

Comment:
this would work but it not intuitive in my opinion:

```diff
diff --git a/apps/calendar/js/bridge.js b/apps/calendar/js/bridge.js
index ae6a4ec..3955085 100644
--- a/apps/calendar/js/bridge.js
+++ b/apps/calendar/js/bridge.js
@@ -6,6 +6,7 @@ var co = require('ext/co');
 var core = require('core');
 var dayObserver = require('day_observer');
 var nextTick = require('common/next_tick');
+var object = require('common/object');
 
 /**
  * Fetch all the data needed to display the busytime information on the event
@@ -42,15 +43,14 @@ exports.observeCalendars = function() {
     // calendarStore.all() returns an object! we convert into an array since
     // that is easier to render/manipulate
     var calendars = yield calendarStore.all();
-    var data = [];
-    for (var id in calendars) {
-      var calendar = calendars[id];
-      var provider = yield calendarStore.providerFor(calendar);
-      data.push({
-        calendar: calendar,
-        capabilities: provider.calendarCapabilities(calendar)
+    var data = yield object.map(calendars, (id, calendar) => {
+      return calendarStore.providerFor(calendar).then(provider => {
+        return {
+          calendar: calendar,
+          capabilities: provider.calendarCapabilities(calendar)
+        };
       });
-    }
+    });
     stream.write(data);
   });
```

note that we need to call `then` and later we return an object... one might think that `data` is an array of promises and not an array of objects...